### ISSUE TITLE
OpenAI o1 compatibility issue

### ISSUE THREAD

**Issue by preritdas** [saoudrizwan/claude-dev#284]
```bash
400 Unsupported value: 'messages[0].role' does not support 'system' with this model.
```

**Comment by shaggy2626** [comment]
im having the same issue. System prompt needs to be removed from the payload. Can't use it for the time being. Hopefully they will update

**Comment by saoudrizwan** [comment]
v1.6.8 now has o1 support! It turns out o1 doesnt natively support system prompt or tool use, so I had to use structured output prompting to achieve the same effect. I hit my rate limit early so couldn't test much, but figured i should put it out there for others to try.

**Comment by preritdas** [comment]
> v1.6.8 now has o1 support! It turns out o1 doesnt natively support system prompt or tool use, so I had to use structured output prompting to achieve the same effect. I hit my rate limit early so couldn't test much, but figured i should put it out there for others to try.

Killer! Thank you sir.

**Comment by preritdas** [comment]
Unfortunately still not quite running appropriately:

```bash
400 Unsupported parameter: 'max_tokens' is not supported with this model. Use 'max_completion_tokens' instead.
```

**Comment by saoudrizwan** [comment]
Are you using OpenAI or OpenRouter? Because I don't have a high enough tier on OpenAI to test this properly. It seems like other projects are seeing this error too, so I think OpenAI updated their API to have a new max_completion_tokens param (since there's now the concept of 'reasoning tokens')

**Comment by saoudrizwan** [comment]
I was right, will update ASAP https://github.com/openai/openai-node/blob/328f75caef55f2d7a829d3de520b741855e2e104/src/resources/chat/completions.ts#L796

**Comment by saoudrizwan** [comment]
Fixed in v1.6.10 (I think, pls confirm)

**Comment by preritdas** [comment]
> Fixed in v1.6.10 (I think, pls confirm)

Confirming! Works. Thank you!
